src/plus/integrations/providers/gitlab.ts

```diff
@@ -157,19 +151,31 @@ export class GitLabIntegration extends ProviderIntegration<GitLabRepositoryDescr
 	}
 }
 
-export class GitLabSelfHostedIntegration extends GitLabIntegration {
-	override readonly authProvider = enterpriseAuthProvider;
-	override readonly id = ProviderId.GitHubEnterprise;
-	override readonly name = 'GitLab Self-Hosted';
-	override get domain(): string {
+export class GitLabIntegration extends GitLabIntegrationBase<HostedProviderId.GitLab> {
+	readonly authProvider = authProvider;
+	readonly id = HostedProviderId.GitLab;
+	protected readonly key = this.id;
+	readonly name: string = 'GitLab';
+	get domain(): string {
+		return metadata.domain;
+	}
+
+	protected get apiBaseUrl(): string {
+		return 'https://gitlab.com/api';
+	}
+}
+
+export class GitLabSelfHostedIntegration extends GitLabIntegrationBase<SelfHostedProviderId.GitHubEnterprise> {
+	readonly authProvider = enterpriseAuthProvider;
+	readonly id = SelfHostedProviderId.GitHubEnterprise;
```

Comment:
This should be `GitLabEnterprise`

Reply:
Please feel free to commit and take over this.

Reply:
Roger that 👍🏼